runtime/src/statistics.rs

```diff
@@ -0,0 +1,51 @@
+use crate::balances;
+
+use rstd::vec::Vec;
+use srml_support::{decl_module, decl_storage};
+
+type AssetId = Vec<u8>;
+type Counter = u64;
+
+pub trait Trait: balances::Trait {}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as statistics {
+        pub InvestorCountPerAsset get(investor_count_per_asset): map AssetId => Counter ;
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+    }
+}
+
+impl<T: Trait> Module<T> {
+    /// It updates our statistics after transfer execution.
+    /// The following counters could be updated:
+    ///     - *Investor count per asset*.
+    ///
+    pub fn update_transfer_stats(
+        ticker: &Vec<u8>,
+        updated_from_balance: T::Balance,
+        updated_to_balance: T::Balance,
+        amount: T::Balance,
+    ) {
+        // 1. Investor count per asset.
+        if amount != 0u128.into() {
+            let counter = Self::investor_count_per_asset(ticker);
+            let mut new_counter = counter;
+
+            if updated_from_balance == 0u128.into() {
+                new_counter = new_counter.checked_sub(1).unwrap_or(new_counter);
+            }
+            if updated_to_balance == amount {
+                new_counter = new_counter.checked_add(1).unwrap_or(new_counter);
+            }
```

Comment:
Need to also check that `amount != 0` because 0 amount transfers are also possible.

Reply:
yes, it is done in line 34.